syntax = "proto3";

package rayclone;

service RemoteRPC{
  rpc RayRemote(RemoteRequest) returns (RemoteResponse){}
}

message RemoteRequest{
  string function_name = 1;
  repeated bytes function_args = 2;
}

message RemoteResponse{
  bytes response = 1; // msgpack buffer containing the response
}
